From 05374595b97eca313c09184cb3eafd3d36c6acdc Mon Sep 17 00:00:00 2001
From: "Wang, ArvinX" <arvinx.wang@intel.com>
Date: Tue, 22 May 2018 19:18:24 +0800
Subject: [PATCH] Modify the LensResolverActivity intent with right component

The music can't play due to the LensResolverActivity always
use the default intent to start activity, correct it using the
ResolveInfo component.

Change-Id: Id760c0e94857cda8659b3f3bcc5ae8359370884e
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-63290
Signed-off-by: Wang, ArvinX <arvinx.wang@intel.com>
---
 .../android/support/car/lenspicker/LensResolverActivity.java   | 10 ++++++++--
 src/com/android/support/car/lenspicker/ResolverAdapter.java    |  2 +-
 2 files changed, 9 insertions(+), 3 deletions(-)

diff --git a/src/com/android/support/car/lenspicker/LensResolverActivity.java b/src/com/android/support/car/lenspicker/LensResolverActivity.java
index bd69a0d..2f4b313 100644
--- a/src/com/android/support/car/lenspicker/LensResolverActivity.java
+++ b/src/com/android/support/car/lenspicker/LensResolverActivity.java
@@ -166,8 +166,14 @@ public class LensResolverActivity extends Activity implements
      * {@link #mResolveIntent}.
      */
     @Override
-    public void onActivitySelected(ResolveInfo info, LensPickerItem item) {
-        ComponentName component = item.getLaunchIntent().getComponent();
+    public void onActivitySelected(ResolveInfo info, LensPickerItem item /*not use*/) {
+        ComponentName component = info.getComponentInfo().getComponentName();
+
+        if (LensPickerUtils.isMediaService(info)) {
+            Intent intent = LensPickerUtils.getMediaLaunchIntent(getPackageManager(), LensPickerUtils.getPackageName(info),
+                    info.serviceInfo.name);
+            component = intent.getComponent();
+        }
 
         if (mAlwaysCheckbox.isChecked()) {
             PackageManager pm = getPackageManager();
diff --git a/src/com/android/support/car/lenspicker/ResolverAdapter.java b/src/com/android/support/car/lenspicker/ResolverAdapter.java
index d9324c8..9fc4fbd 100644
--- a/src/com/android/support/car/lenspicker/ResolverAdapter.java
+++ b/src/com/android/support/car/lenspicker/ResolverAdapter.java
@@ -126,7 +126,7 @@ public class ResolverAdapter extends RecyclerView.Adapter<ResolverListRow>
                     }
 
                     items.add(new LensPickerItem(displayName,
-                            getComponentIcon(packageManager, aInfo), launchIntent,
+                            getComponentIcon(packageManager, aInfo), null /* launchIntent */,
                             null /* facetId */));
                 } catch (PackageManager.NameNotFoundException e) {
                     // skip this package.
-- 
1.9.1

